R/verb-count.R

```diff
@@ -26,6 +27,24 @@ count.tbl_lazy <- function(x, ..., wt = NULL, sort = FALSE, name = NULL) {
   out
 }
 
+#' @rdname count.tbl_lazy
+#' @importFrom dplyr add_count
+#' @importFrom lifecycle deprecated
+#' @export
+add_count.tbl_lazy <- function (x, ..., wt = NULL, sort = FALSE, name = NULL, .drop = deprecated()) {
+  if (!missing(.drop)) {
+    abort("`.drop` argument not supported for lazy tables.")
+  }
+  if (!missing(...)) {
+    out <- group_by(x, ..., .add = TRUE)
+  }
+  else {
+    out <- x
+  }
+  out <- dplyr::add_tally(out, wt = !!enquo(wt), sort = sort, name = name)
+  group_by(out, !!!syms(group_vars(x)))
```

Comment:
Should we regroup only if we grouped before?

Reply:
* grouped before -> grouped again
* not grouped before -> this boils down to `group_by(out)` which adds an unnecessary ungroup operation but still produces the correct result.

One could add an extra check before it the tbl was grouped but I think that `group_by()` resp. `ungroup()` should avoid adding the unnecessary ungroup operation.